services/grafana/client.go

```diff
@@ -202,22 +211,55 @@ func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, er
 		// check only default organization (with ID 1)
 		if id, _ := m["orgId"].(float64); id == 1 {
 			role, _ := m["role"].(string)
-			switch role {
-			case "Viewer":
-				return viewer, nil
-			case "Editor":
-				return editor, nil
-			case "Admin":
-				return admin, nil
-			default:
-				return none, nil
-			}
+			return c.convertRole(role), nil
 		}
 	}
 
 	return none, nil
 }
 
+func (c *Client) isAPIKeyAuth(authHeader string) bool {
+	switch {
+	case strings.Contains(authHeader, "Bearer"):
+		return true
+	case strings.Contains(authHeader, "Basic"):
+		h := strings.TrimPrefix(authHeader, "Basic")
+		d, err := base64.StdEncoding.DecodeString(strings.TrimSpace(h))
+		if err != nil {
+			return false
+		}
+		return strings.HasPrefix(string(d), "api_key:")
+	}
+	return false
+}
+
+func (c *Client) convertRole(role string) role {
+	switch role {
+	case "Viewer":
+		return viewer
+	case "Editor":
+		return editor
+	case "Admin":
+		return admin
+	default:
+		return none
+	}
+}
+
+func (c *Client) getRoleForAPIKey(ctx context.Context, authHeaders http.Header) (role, error) {
+	var k map[string]interface{}
+	if err := c.do(ctx, "GET", "/api/auth/key", "", authHeaders, nil, &k); err != nil {
+		return none, err
+	}
+
+	if id, _ := k["orgId"].(float64); id != 1 {
```

Comment:
What if the Customers use multiple orgs?

Reply:
Now we don't support multiple orgs.
For users we support only default org, so I decided that we should use the same behavior for API keys.

Reply:
https://github.com/percona/pmm-managed/pull/653/files#diff-a9c8e70f3cab071e6bbcb1cc40e5c8a0f5d9bc87cdc48b0fca4e3c70466d12d6R212